core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/DataProcessor.scala

```diff
@@ -69,26 +69,30 @@ class DataProcessor(
     inputGateway.getChannel(channelId).getQueuedCredit
   }
 
-  /** provide API for actor to get stats of this executor
-    *
-    * @return (input tuple count, output tuple count)
+  /**
+    * provide API for actor to get stats of this operator
     */
   def collectStatistics(): WorkerStatistics =
-    statisticsManager.getStatistics(stateManager.getCurrentState, executor)
+    statisticsManager.getStatistics(executor)
 
   /**
     * process currentInputTuple through executor logic.
     * this function is only called by the DP thread.
     */
   private[this] def processInputTuple(tuple: Tuple): Unit = {
     try {
+      val portIdentity: PortIdentity =
+        this.inputGateway.getChannel(inputManager.currentChannelId).getPortId
       outputManager.outputIterator.setTupleOutput(
         executor.processTupleMultiPort(
           tuple,
-          this.inputGateway.getChannel(inputManager.currentChannelId).getPortId.id
+          portIdentity.id
         )
       )
-      statisticsManager.increaseInputTupleCount()
+
+      if (!portIdentity.internal) {
+        statisticsManager.increaseInputTupleCount(portIdentity.id)
+      }
```

Comment:
can we also collect the internal port statistics in `StatisticsManager`, for debugging purposes? It's fine if we do not report those numbers to the outside (e.g., to the controller), but keeping this information locally can help.

Reply:
I'll fix this in the different PR